### PR TITLE
DUPLO-16126 DUPLO-17556 DUPLO-17573 DUPLO-17479 dynamodb update

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Added support for local secondary index  -DUPLO-17573
Fixed Updating issue for point_in_time_recovery and delete_protection DUPLO-17556, 
Fixed Updating of global secondary index with support of actions(create,update,delete), server side encryption DUPLO-17573 and DUPLO-16721
Doc update for attribute_definition and key_schema DUPLO-17479
## Summary of changes

This PR does the following:

- Fixed update issues
- Implemented support for local secondary index
- Updated docs

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-16126 (v0.10.13/v0.10.14/develop)TF: Not able to update dynamodb_table_v2 resource.](https://app.clickup.com/t/8655600/DUPLO-16126)
Task linked: [DUPLO-17479 TF:Doc: Make 'attribute' and 'key_schema' as required field for resource 'duplocloud_aws_dynamodb_table_v2'.](https://app.clickup.com/t/8655600/DUPLO-17479)
Task linked: [DUPLO-17556 (develop)TF: Not able to update dynamodb_table_v2 resource.](https://app.clickup.com/t/8655600/DUPLO-17556)
Task linked: [DUPLO-17573 TF-support for dynamodb : add local_secondary_index support](https://app.clickup.com/t/8655600/DUPLO-17573)

File: duplosdk/client.go [hunk 1]
```diff
@@ -221,7 +221,7 @@ func (c *Client) doAPI(verb string, apiName string, apiPath string, rp interface
 func (c *Client) doAPIWithRequestBody(verb string, apiName string, apiPath string, rq interface{}, rp interface{}) ClientError {
 	apiName = fmt.Sprintf("%sAPI %s", strings.ToLower(verb), apiName)
 	url := fmt.Sprintf("%s/%s", c.HostURL, apiPath)
-
+	log.Printf("REQUEST NODY BEFORE SERIALIZE %+v", rq)
```
Comment (tahir-duplo) [inline duplosdk/client.go:224]: Remove this log

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-16721 TF: duplocloud_aws_dynamodb_table_v2: Updating `global_secondary_index` gives success message but the data is not being updated.](https://app.clickup.com/t/8655600/DUPLO-16721)

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**